great_expectations/core/expectation_configuration.py

```diff
@@ -1417,7 +1417,9 @@ def get_domain_type(self) -> Optional[MetricDomainTypes]:
         if "column_list" in self.kwargs:
             return MetricDomainTypes.MULTICOLUMN
 
-        return None
+        raise ValueError(
+            'Unable to determine "domain_type" of this "ExpectationConfiguration" object from "kwargs" and heuristics.'
```

Comment:
You might want to print the values of `expectations_types` and `kwargs` here so we know what the bad object looks like for debugging.